superset-frontend/src/explore/controlPanels/DeckArc.js

```diff
@@ -26,44 +41,98 @@ export default {
       label: t('Query'),
       expanded: true,
       controlSetRows: [
-        ['start_spatial', 'end_spatial'],
-        ['row_limit', 'filter_nulls'],
+        [
+          {
+            name: 'start_spatial',
+            config: {
+              type: 'SpatialControl',
+              label: t('Start Longitude & Latitude'),
+              validators: [nonEmpty],
+              description: t('Point to your spatial columns'),
+              mapStateToProps: state => ({
+                choices: columnChoices(state.datasource),
+              }),
+            },
+          },
+          {
+            name: 'end_spatial',
+            config: {
+              type: 'SpatialControl',
+              label: t('End Longitude & Latitude'),
+              validators: [nonEmpty],
+              description: t('Point to your spatial columns'),
+              mapStateToProps: state => ({
+                choices: columnChoices(state.datasource),
+              }),
+            },
+          },
+        ],
+        ['row_limit', filterNulls],
         ['adhoc_filters'],
       ],
     },
     {
       label: t('Map'),
       controlSetRows: [
-        ['mapbox_style', 'viewport'],
-        ['autozoom', null],
+        ['mapbox_style', viewport],
+        [autozoom, null],
       ],
     },
     {
       label: t('Arc'),
       controlSetRows: [
-        ['color_picker', 'target_color_picker'],
-        ['dimension', 'color_scheme', 'label_colors'],
-        ['stroke_width', 'legend_position'],
-        ['legend_format', null],
+        [
+          'color_picker',
+          {
+            name: 'target_color_picker',
+            config: {
+              label: t('Target Color'),
+              description: t('Color of the target location'),
+              type: 'ColorPickerControl',
+              default: PRIMARY_COLOR,
+              renderTrigger: true,
+            },
+          },
+        ],
+        [
+          Object.assign({}, dimension, {
```

Comment:
could do `{...dimension, { label: ... } }`

Reply:
Agreed on this direction. I quickly noticed this happens all over the place (maybe I'm not the only one with old habits!). Added an eslint rule to lint-fix this in another PR, #9466